src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,777 @@
+
+1. Define a POJO `Book` object
+
+```java
+class Book {
+    private String title;
+
+    public Book(String title) {
+        this.title = title;
+    }
+}
+```
+
+2. Define a fetcher
+
+```java
+class BookFetcher implements DataFetcher<Book> {
+    @Override
+    Book get(DataFetchingEnvironment env) {
+        return new Book("Reindeers World");
+    }
+}
+```
+
+The book is hardcoded for the example's sake. Normaly it would probably call some 3rd party API. But as far as GraphQL is concerned, it doesn't matter as long as the fetcher returns a `Book` object. 
+
+3. Assign the fetcher to operation
+
+```java
+//...
+RuntimeWiring runtimeWiring = newRuntimeWiring()
+    .type("QueryType", builder -> builder.dataFetcher("book", new BookFetcher()))
+    .build()
+//then load the schema file and construct GraphQL object
+```
+
+You'll end up with a `graphQL` object (of class `GraphQL`). You can then ask this object to process a query, e.g:
+
+```graphql
+{
+    book {
+        title
+    }
+}
+```
+
+And you should get the following output:
+
+```json
+{
+    "data": {
+        "book": {
+            "title": "Reindeers World"
+        }
+    }
+}
+```
+
+Quite simple.
+
+## Vert.x Web GraphQL module
+Note that the above java implemention has nothing to do with HTTP. This is where the Vert.x implementation comes in handy. Vert.x provides an out of the box HTTP handler for GraphQL. We just need to provide it with a configured `GraphQL` object and it's ready to serve requests through HTTP.
+
+## Get the best out of GraphQL and Knot.x
+There are two steps to integrate GraphQL and Knot.x. The first is to define a GraphQL handler factory that will return a `GraphQLHandler` (the one from the Vert.x implementation). It will allow us to define and configure GraphQL endpoint in the Knot.x configuration.
+
+We could stop at implementing the factory but we wouldn't be using the full power of Knot.x. Knot.x has some great mechanisms for _fetching_ data from 3rd parties. Therefore, the second step is to create data fetchers that make use of Knot.x integration functionalities.
+
+> It's assumed you have the project set up and ready for development. If you have troubles with that check the [Getting Started tutorials](/tutorials/getting-started-with-knotx-stack/2_0/).
+
+## Step 0 - Google Books API, schema and model
+Let's stick with books. We'll implement a sample service serving books from Google Books API. Google Books is a typical REST API. It returns a lot of information. In this tutorial we'll extract just a title, publisher and list of authors (google api returns authors as a list of strings). The response from the Google API looks like this:
+
+```json5

```

Comment:
```suggestion
```json
```

Reply:
It's on purpose. JSON5 is an extension of JSON. It supports comments that I use in this snippet. Knotx website treats it as a regular JSON, but github for instance supports JSON5.

So yea, it's a json5 snippet